packages/management-api/src/managed-pod.ts

```diff
@@ -117,36 +160,60 @@ export class ManagedPod {
       })
         .done((data: string, textStatus: string, xhr: JQueryXHR) => {
           if (xhr.status !== 200) {
-            reject()
+            this.setManagementError(xhr.status, textStatus)
+            reject(this.getManagementError())
             return
           }
 
+          let jsonResponse
           try {
-            const resp = JSON.parse(data)
-            if ('error' in resp) {
-              throw new Error(resp.error)
-            }
-
-            if ('value' in resp && 'agent' in resp.value) {
-              log.debug('Found jolokia agent at:', this.jolokiaPath, 'version:', resp.value.agent)
-              resolve(this.jolokiaPath)
-              return
-            } else {
-              throw new Error('Detected jolokia but cannot determine agent or version')
-            }
+            jsonResponse = JSON.parse(data)
```

Comment:
Basically, if we know it's a response from Jolokia we should use the type provided by `jolokia.js`. We are modern code.
https://github.com/jolokia/jolokia/blob/af7075546281f65da13cd42b075bc03670f2535d/client/javascript/src/main/javascript/index.d.ts#L504-L516